cmake/checks/check_02_system_features.cmake

```diff
@@ -106,51 +106,6 @@ ENDIF()
 ########################################################################
 
 
-IF( CMAKE_SYSTEM_NAME MATCHES "CYGWIN"
```

Comment:
What happens with this change if you configure with cygwin or mingw? Can we just blacklist them with a helpful error message as in "not supported, try MSVC if you are brave"?